Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitInspector.cs

```diff
@@ -118,11 +138,177 @@ public override void OnInspectorGUI()
+                #region draw search field input
+
+                using (new EditorGUILayout.HorizontalScope())
+                {
+                    EditorGUILayout.LabelField("Search Profile For: ", GUILayout.MaxWidth(120));
+                    config.SearchFieldString = EditorGUILayout.TextField(SessionState.GetString(searchDisplaySearchFieldKey, string.Empty), GUILayout.ExpandWidth(true));
+                    if (GUILayout.Button("Clear", EditorStyles.miniButton, GUILayout.MaxWidth(50)))
+                    {
+                        config.SearchFieldString = string.Empty;
+                    }
+                }
+
+                config.RequireAllKeywords = SessionState.GetBool(searchDisplayRequireAllKeywordsKey, true);
+                config.SearchTooltips = SessionState.GetBool(searchDisplaySearchTooltipsKey, true);
+                config.SearchFieldObjectNames = SessionState.GetBool(searchDisplaySearchFieldObjectNamesKey, false);
+                config.SearchChildProperties = SessionState.GetBool(searchDisplaySearchChildPropertiesKey, true);
+
+                #endregion
+
+                if (!string.IsNullOrEmpty(config.SearchFieldString))
+                {
+                    // For search, draw our custom search GUI
+                    DrawSearchGUI(activeProfile.objectReferenceValue);
+                }
+                else
+                {
+                    // For configure, show the default inspector GUI
+                    UnityEditor.Editor activeProfileEditor = CreateEditor(activeProfile.objectReferenceValue);
+                    activeProfileEditor.OnInspectorGUI();
+                }
+
+                #region store search settings
+
+                SessionState.SetString(searchDisplaySearchFieldKey, config.SearchFieldString);
+                SessionState.SetBool(searchDisplayRequireAllKeywordsKey, config.RequireAllKeywords);
+                SessionState.SetBool(searchDisplaySearchTooltipsKey, config.SearchTooltips);
+                SessionState.SetBool(searchDisplaySearchFieldObjectNamesKey, config.SearchFieldObjectNames);
+                SessionState.SetBool(searchDisplaySearchChildPropertiesKey, config.SearchChildProperties);
+
+                #endregion
             }
         }
 
+        #region search
+
+        private void DrawSearchGUI(UnityEngine.Object activeProfileObject)
+        {
+            #region options
+
+            bool optionsFoldout = SessionState.GetBool(searchDisplayOptionsFoldoutKey, false);
+            optionsFoldout = EditorGUILayout.Foldout(optionsFoldout, "Search Preferences", true);
+            SessionState.SetBool(searchDisplayOptionsFoldoutKey, optionsFoldout);
+            if (optionsFoldout)
+            {
+                config.RequireAllKeywords = EditorGUILayout.Toggle("Require All Keywords", config.RequireAllKeywords);
+                config.SearchTooltips = EditorGUILayout.Toggle("Search Tooltips", config.SearchTooltips);
```

Comment:
No tooltips. Not exaclty clear what child properties and gameobject names are considered